src/meta/src/hummock/compaction/tier_compaction_picker.rs

```diff
@@ -0,0 +1,234 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use bytes::Bytes;
+use risingwave_hummock_sdk::key::{user_key, FullKey};
+use risingwave_hummock_sdk::key_range::KeyRange;
+use risingwave_hummock_sdk::HummockEpoch;
+use risingwave_pb::hummock::{Level, LevelType, SstableInfo};
+
+use super::SearchResult;
+use crate::hummock::level_handler::LevelHandler;
+
+const DEFAULT_MAX_COMPACTION_BYTES: u64 = 2 * 1024 * 1024 * 1024; // 2GB
+const DEFAULT_LEVEL0_MAX_FILE_NUMBER: usize = 16;
+const DEFAULT_LEVEL0_TRIGGER_NUMBER: usize = 4;
+
+#[derive(Debug)]
+pub struct TierCompactionPicker {
+    compact_task_id: u64,
+    max_compaction_bytes: u64,
+    level0_max_file_number: usize,
+    level0_trigger_number: usize,
+}
+
+impl TierCompactionPicker {
+    pub fn new(compact_task_id: u64) -> TierCompactionPicker {
+        TierCompactionPicker {
+            compact_task_id,
+            // TODO: set it by cluster configure or vertical group configure.
+            max_compaction_bytes: DEFAULT_MAX_COMPACTION_BYTES,
+            level0_max_file_number: DEFAULT_LEVEL0_MAX_FILE_NUMBER,
+            level0_trigger_number: DEFAULT_LEVEL0_TRIGGER_NUMBER,
+        }
+    }
+
+    pub fn pick_compaction(
+        &mut self,
+        levels: Vec<Level>,
+        level_handlers: &mut [LevelHandler],
+    ) -> Option<SearchResult> {
+        let select_level = 0;
+        let next_task_id = self.compact_task_id;
+        if levels[0].table_infos.is_empty() {
```

Comment:
why 0 but not `select_level` here?

Reply:
It's a tier_compaction_picker, not leveled_compaction_picker.
tier_compaction_picker only supports compact files from L0 to L1.